ni_measurementlink_service/_dotenvpath.py

```diff
@@ -44,16 +44,35 @@ def _get_caller_path() -> Optional[Path]:
     for frame, _ in traceback.walk_stack(inspect.currentframe()):
         if frame.f_code.co_filename:
             module_path = Path(frame.f_code.co_filename)
-            if module_path.exists() and not _is_relative_to(module_path, nims_path):
+            if _exists(module_path) and not _is_relative_to(module_path, nims_path):
                 return module_path
 
     return None
 
 
-def _is_relative_to(path: PurePath, other: PurePath) -> bool:
-    if sys.version_info >= (3, 9):
+# Path.exists() throws OSError when the path has invalid file characters.
+# https://github.com/python/cpython/issues/79487
+if sys.version_info >= (3, 10):
+
+    def _exists(path: Path) -> bool:
```

Comment:
Used `Path` instead of `PurePath` because mypy throwed an error: "PurePath" has no attribute "exists".

Reply:
Right, `exists` does need to access the filesystem :)